labs/04_play_with_playwright/test_product.py

```diff
@@ -1,8 +1,36 @@
 import re
 from playwright.sync_api import Page, expect
 
+APP_URL = "http://localhost:5173"
 
-def test_has_title(page: Page):
-    page.goto("https://playwright.dev/")
-    expect(page).to_have_title(re.compile("Playwright"))
-    expect(page.get_by_text("Get started")).to_be_visible()
\ No newline at end of file
+
+def test_admin_add_product(page: Page):
+    page.goto(APP_URL)
+
+    # Login as admin
+    page.fill('//input[@placeholder="Username"]', "admin")
+    page.fill('//input[@placeholder="Password"]', "1234")
+    page.click('//button[normalize-space()="Login"]')
+
+    # Add product
+    page.fill('//input[@placeholder="Product Name"]', "IT course")
+    page.click('//button[normalize-space()="Create Product"]')
+
+    # Assert: product is visible in the app
+    expect(page.locator("body")).to_contain_text("IT course")
```

Comment:
Here you followed the same body validation made in selenium. Remember that this could make your test flaky due to false positives.